src/main/java/de/tum/in/www1/artemis/service/ProgrammingSubmissionService.java

```diff
@@ -1,8 +1,11 @@
 package de.tum.in.www1.artemis.service;
 
+import java.net.URL;
 import java.time.ZonedDateTime;
 import java.util.Optional;
 
+import org.eclipse.jgit.api.errors.GitAPIException;
```

Comment:
I was wondering whether we should really expose these classes outside of the GitService. I think from a system design point of view (low coupling and high cohesion), it would be better, if only the GitService has dependencies to org.eclipse.jgit.* if possible.

Reply:
I agree; the issue is with the way the GitService was implemented it exposes the GitApiException in its methods instead of providing more information on what the actual issue is or means.

I will change it for the `getLastCommitHash` method.

Reply:
Done.